ts/parts/Axis.ts

```diff
@@ -5785,6 +5812,41 @@ extend(Axis.prototype, /** @lends Highcharts.Axis.prototype */{
             axis.cleanStacks();
         }
 
+        // Extend the Axis to calculate start min/max values
```

Comment:
What is the purpose of this block? Why not just use `axis.dataMin` and `axis.dataMax`? They are available at this point: https://jsfiddle.net/highcharts/65L208qv/.

Reply:
The `panningState` object is required, because there is the difference of how **getExtremes()** function works on x and y axis, and we need to store initial data min and max values.

Let us take a situation, when a user zoomed the chart or set initial extremes on axis. In case of xAxis, `getExtremes` returns the `dataMin` and `dataMax` values, which are actual data extreme values of series related with this axis. However, after calling `getExtremes()` on yAxis, we're always getting values of currently visible range (excluding all data points located out of the current visible axis range).

https://jsfiddle.net/BlackLabel/ca0qku1x/

Reply:
Thanks for the explanation. I will try to refactor `Series.getExtremes` to make it return the data extremes, then we can call it here instead of recreating the series data loop. This is something we discussed earlier too, that the `getExtremes` function should return values rather than setting them on the instance.